src/storage/local-project-storage-client.ts

```diff
@@ -0,0 +1,118 @@
+// Implementation of ProjectStorageClient backed by IndexedDB,
+// for storing play project data and files locally in the browser.
+//
+// This is fairly primitive, and doesn't bother with relational data ---
+// it simply stores the entire Project objects, ProjectFiles attached.
+
+import type {
+  PlayProject,
+  ProjectStorageClient
+} from './project-storage-client.js'
+
+const DB_NAME = 'PlayProjectDB'
+const DB_VERSION = 1
+const PROJECT_STORE = 'projects'
+
+function openDB(): Promise<IDBDatabase> {
+  return new Promise((resolve, reject) => {
+    const request = indexedDB.open(DB_NAME, DB_VERSION)
+
+    request.onupgradeneeded = () => {
+      const db = request.result
+      if (!db.objectStoreNames.contains(PROJECT_STORE)) {
+        db.createObjectStore(PROJECT_STORE, {keyPath: 'id'})
+      }
+    }
+
+    request.onsuccess = () => {
+      resolve(request.result)
+    }
+
+    request.onerror = () => {
+      reject(request.error)
+    }
+  })
+}
+
+/** Browser-local implementation of ProjectStorageClient. */
+export class LocalProjectStorageClient implements ProjectStorageClient {
+  async CreateProject(name: string): Promise<PlayProject> {
+    const db = await openDB()
+    const transaction = db.transaction([PROJECT_STORE], 'readwrite')
+    const store = transaction.objectStore(PROJECT_STORE)
+
+    const id = crypto.randomUUID()
+    const project: PlayProject = {
+      id,
+      name,
+      createdAt: new Date(),
+      updatedAt: new Date(),
+      authorId: '',
+      files: []
+    }
+
+    return new Promise((resolve, reject) => {
+      const request = store.add(project)
+      request.onsuccess = () => {
+        resolve(project)
+      }
+      request.onerror = () => {
+        reject(request.error)
+      }
+    })
```

Comment:
I wonder if we can create a helper for that (and other similar operations)

```ts
function IDBRequestToPromise<SuccessPayload>(request: IDBRequest, successPayload: SuccessPayload) {
return new Promise((resolve, reject) => {
      request.onsuccess = () => {
        resolve(successPayload)
      }
      request.onerror = () => {
        reject(request.error)
      }
    })
}
```

```ts
return IDBRequestToPromise<Project>(store.add(project), project);
```

Something like that. IDK if it's better or not for readability though.